beacon-chain/blockchain/head.go

```diff
@@ -159,7 +161,7 @@ func (s *Service) setHead(root [32]byte, block *ethpb.SignedBeaconBlock, state *
 		slot:  block.Block.Slot,
 		root:  root,
 		block: stateTrie.CopySignedBeaconBlock(block),
-		state: state.Copy(),
+		state: state,
```

Comment:
This doesn't feel very safe because `setHead` is called when:
1.) initial sync
2.) post initial sync

I agree for 1.) no `Copy` is ok.  
But for 2.) I do think we need the `Copy` or we'll run into issues. I believe that is why the Copy is added there to begin with. 

A mitigation is to have `setHeadInitialSync` and `setHeadPostSync`

Reply:
That makes sense I can change it to only account for initial sync. Although I did look at the code path in regular sync and after setting the head, the state isn't used anywhere else. But to reduce the possibility of accidentally breaking anything, I will change it back.

Reply:
Alright done